lib/cog/bundle.rb

```diff
@@ -1,58 +1,49 @@
 class Cog
   class Bundle
-    attr_reader :name, :config
+    attr_reader :name
 
-    def initialize(name, base_dir: nil, config_file: nil)
+    def initialize(name, base_dir: nil)
       @name = name
       @base_dir = base_dir || File.dirname($0)
-      @config = load_config(config_file || File.join(@base_dir, 'config.yaml'))
       @module = create_bundle_module
-
-      load_commands
     end
 
     def create_bundle_module
       return if Object.const_defined?('CogCmd')
-
       Object.const_set('CogCmd', Module.new)
       CogCmd.const_set(@name.capitalize, Module.new)
     end
 
-    def load_config(path=nil)
-      path ||= File.join(@base_dir, 'config.yaml')
-      Cog::Config.new(path)
-    end
-
-    def load_commands
-      @config[:commands].each do |command, config|
-        command_path = command.split('-')
-        require File.join(@base_dir, 'lib', 'cog_cmd', @name, *command_path)
-      end
-    end
-
-    def run_command
-      command = ENV['COG_COMMAND']
+    def command_instance(command_name)
+      command_path = command_name.split('-')
+      require File.join(@base_dir, 'lib', 'cog_cmd', @name, *command_path)
 
       # translate snake-case command names to camel case
-      command_class = command.gsub(/(\A|_)([a-z])/) { $2.upcase }
+      command_class = command_name.gsub(/(\A|_)([a-z])/) { $2.upcase }
```

Comment:
Unrelated to the PR, but there have been other places where I've wanted ActiveSupport. We should probably just include it and be done with it.